### PR TITLE
Provisioner flavor fix

### PR DESCRIPTION
**Pull request by efikalti**
Provisioner checks for allow creation in each flavor before creating the vm with the specified resources.

### CODE REVIEW

File: core/fokia/provisioner.py [hunk 1]
```diff
@@ -492,6 +500,16 @@ def get_server_private_ip(self, server_id):
     """
     CHECK RESOURCES
     """
+    def check_flavor(vcpus, ram, disk):
+        """
+        Check if flavor with this resources allows creation.
```
Comment (skanct) [inline core/fokia/provisioner.py]: Can you please rewrite the description? It is difficult to understand what this method is supposed to do

Comment (efikalti) [inline core/fokia/provisioner.py]: I added a more detailed description to the method,please review @skanct

File: core/fokia/provisioner.py [hunk 2]
```diff
@@ -156,6 +156,14 @@ def create_lambda_cluster(self, vm_name, **kwargs):
                                               project_name=kwargs['project_name'])
 
         if response:
+            # Check flavors for master and slaves
+            if not check_flavor(vcpus=kwargs['vcpus_master'], ram=kwargs['ram_master'],disk=kwargs['disk_master']):
```
Comment (makistsantekidis) [inline core/fokia/provisioner.py]: `if not self.check_flavor(.........)`

Comment (efikalti) [inline core/fokia/provisioner.py]: fixed in previous commit

**Comment by makistsantekidis** [comment]
+1

**Pull request merged**